### PR TITLE
 packages added: Paho mqtt C and Paho mqtt C++

### PR DESCRIPTION
**Pull request by Altyrost**

### CODE REVIEW

File: packages/p/pahomqttc/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                         "-DPAHO_BUILD_DOCUMENTATION=FALSE"}
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        import("package.tools.cmake").install(package, configs)
+    end)
+
+    on_test(function (package)
+        assert(package:has_cxxtypes("MQTTClient_message", {includes = "MQTTClient.h"}))
+    end)   
+package_end()
```
Comment (waruqi) [inline packages/p/pahomqttc/xmake.lua:25]: please remove package_end

File: packages/p/pahomqttcpp/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        table.insert(configs, "-DPAHO_MQTT_C_INCLUDE_DIRS=" .. pahomqttc:installdir("include"))
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        import("package.tools.cmake").install(package, configs)
+    end)
+
+    on_test(function (package)
+        assert(package:has_cxxtypes("mqtt::message", {includes = "mqtt/message.h"}))
+    end)    
+package_end()
```
Comment (waruqi) [inline packages/p/pahomqttcpp/xmake.lua:27]: and here

File: packages/p/pahomqttcpp/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        local configs = {"-DCMAKE_INSTALL_PREFIX=" .. pahomqttc:installdir()}
+        table.insert(configs, "-DPAHO_MQTT_C_LIBRARIES=" .. pahomqttc:installdir("lib"))
+        table.insert(configs, "-DPAHO_MQTT_C_INCLUDE_DIRS=" .. pahomqttc:installdir("include"))
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        import("package.tools.cmake").install(package, configs)
+    end)
+
+    on_test(function (package)
+        assert(package:has_cxxtypes("mqtt::message", {includes = "mqtt/message.h"}))
```
Comment (waruqi) [inline packages/p/pahomqttcpp/xmake.lua:25]: please use check_cxxsnippets or has_cxxfuncs to check c++ symbols links, you can see other packages

File: packages/p/pahomqttc/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        local configs = {"-DPAHO_BUILD_SAMPLES=FALSE",
+                         "-DPAHO_WITH_SSL=FALSE",
+                         "-DPAHO_BUILD_DOCUMENTATION=FALSE"}
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        import("package.tools.cmake").install(package, configs)
+    end)
+
+    on_test(function (package)
+        assert(package:has_cxxtypes("MQTTClient_message", {includes = "MQTTClient.h"}))
```
Comment (waruqi) [inline packages/p/pahomqttc/xmake.lua:23]: is this c library? you can use has_cfuncs to check symbol links

File: packages/p/pahomqttcpp/xmake.lua (hunk 3 above)
Comment (Altyrost) [reply packages/p/pahomqttcpp/xmake.lua]: see #[454d0b1](https://github.com/xmake-io/xmake-repo/pull/2757/commits/454d0b1814cec2a4ff0faf230b11613ea2c10e50)

File: packages/p/pahomqttc/xmake.lua (hunk 4 above)
Comment (Altyrost) [reply packages/p/pahomqttc/xmake.lua]: Yes it is a C lib. 

see #[682661b](https://github.com/xmake-io/xmake-repo/pull/2757/commits/682661b996cff7df110718494fa34bd1fc8288e1)

Thank you for the reviews, it's all new to me. I appreciate it.

**Comment by Altyrost** [comment]
I'll work this out on my own before doing a new pull request.

**Pull request closed without merging**